cdap-api/src/main/java/co/cask/cdap/api/data/batch/Input.java

```diff
@@ -206,10 +201,21 @@ private DatasetInput(String name, Map<String, String> arguments, @Nullable Itera
       return arguments;
     }
 
+    public String getNamespace() {
+      return namespace;
+    }
+
     @Nullable
     public List<Split> getSplits() {
       return splits;
     }
+
+    public <T extends Input> T fromNamespace(String namespace) {
```

Comment:
I don't think we need type parameterization here.
Can simply return DatasetInput.